samples/Samples.Mvc5/Controllers/HomeController.cs

```diff
@@ -131,6 +133,35 @@ public ActionResult About()
         /// <returns>The <see cref="ActionResult"/>.</returns>
         public ActionResult ResultsAuthorization() => View();
 
+        public async Task<ActionResult> CallAsync()
+        {
+            var profiler = MiniProfiler.Current;
+            var service = new FooService();
+            using (profiler.Step("action"))
+            {
+                var head = profiler.Head;
+                var tasks = new List<Task>();
+                for (int i = 0; i < 10; i++)
+                {
+                    tasks.Add(service.FooAsync(profiler));
+                }
+                await Task.WhenAll(tasks);
+                return Content("All good");
+            }
+        }
+
+        class FooService
+        {
+            public async Task FooAsync(MiniProfiler profiler)
```

Comment:
why are we passing the profiler into the service? I don't think this is a standard practice. I think it's better if the service just uses Miniprofiler.Current.